iP

ChuaCleon opened the pull request:
My iP progress so far, do feel free to leave any comments for improvement!

Commit:
Remove Duke from existence
Introduce brand-new improved Number 1 Yapper Chatbot aka Yappatron

Commit:
Remove Duke

Commit:
Add echo till user says bye.

Commit:
Add input actions to array
Echo back list when user wants it

Commit:
Add task class for task object as each task has description and status
Mark or unmark tasks when done/undone respectively

Commit:
Create Deadline, Events, Todo class
Extend these classes from Task superclass
Edit main Yappatron class to support these subclasses of Tasks via polymorphism

Commit:
Improve abstraction in main Yappatron class.
Delegate managing of tasks like add, mark, list to new Taskmanager class

YubotKwng commented on src/main/java/Taskmanager.java:
```diff
@@ -0,0 +1,49 @@
+        Yappatron.taskArray[Yappatron.taskNumber++] = new Todo(input.substring(input.indexOf(" ")));
+    }
+
+    public void addDeadline(String input){
```
Good job. Each method in this class is fairly short and focused on a single task, adhering to the guideline of avoiding long methods.

YubotKwng commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,32 @@
+ public class Task {
```
Good job. The class is simple and well-structured.

YubotKwng commented on src/main/java/Yappatron.java:
```diff
@@ -0,0 +1,43 @@
+public class Yappatron {
+    public static final int MAX_TASKS = 100;
+    public static int taskNumber = 0;
+    public static Task[] taskArray = new Task[MAX_TASKS];
```
The use of a fixed-size array (Task[]) reduces flexibility and scalability. Exceeding MAX_TASKS will cause the program to fail, limiting the number of tasks that can be managed.

YubotKwng commented on src/main/java/Yappatron.java:
```diff
@@ -0,0 +1,43 @@
+                else if (input.startsWith("deadline")){
+                    taskManager.addDeadline(input);
+                }
+                else{
```
Having multiple else if conditions can make the code harder to maintain and scale.

Commit:
Handle exceptions from input by users
Create new EmptyTaskEntry that inherit from Exception parent class for empty inputs
Handle incorrect input from users (missing from/to in event or by in deadline)

Commit:
Merge branch 'branch-Level-5'

Commit:
Add deleteTask function to delete task indicated by user
Change from using array to ArrayList for easier addition and deletion
Remove unused variables
Edit functions in TaskManager to add/remove/mark/unmark tasks into ArrayList

Commit:
Add FileManager class to create txt file for storage of data
Load data from txt file everytime program runs.

Commit:
Merge branch 'branch-Level-6'

Commit:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/Yappatron.java

Commit:
Edit output formatting inconsistencies.

Commit:
Move file writing capability from Yappatron to FileManager.
Remove unused variables

okkhoy commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,10 @@
+        super(description);
+        this.deadline = deadline;
+    }
+    public String toString(){
```
introduce vertical spacing for better readability

Jk2987 replied:
ok

okkhoy commented on src/main/java/Events.java:
```diff
@@ -0,0 +1,12 @@
+        this.startTime = startTime;
+        this.endTime = endTime;
+    }
+    public String toString(){
```
spacing issue

okkhoy commented on src/main/java/FileManager.java:
```diff
@@ -0,0 +1,89 @@
+            f.createNewFile();
+        }catch (IOException e){
+            System.out.println("An error occured");
+            e.printStackTrace();
```
printing stack trace is generally not the best thing to do for error handling,

okkhoy commented on src/main/java/FileManager.java:
```diff
@@ -0,0 +1,89 @@
+                if (line.contains("[X]")){
+                    todo.markAsDone();
+                }
+            }
+
+
+            else if (line.startsWith("[D]")){
```
coding standard violation

okkhoy commented on src/main/java/FileManager.java:
```diff
@@ -0,0 +1,89 @@
+    }
+
+
+    public void printFile() throws FileNotFoundException {
```
all non-trivial methods need header comments

okkhoy commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,32 @@
+     public String toString(){
+         if (isDone()){
+             return "[X] " + getDescription();
+         }
+         else{
```
coding standard violation

okkhoy commented on src/main/java/FileManager.java:
```diff
@@ -0,0 +1,89 @@
+    }
+
+    public void writeFile() throws IOException{
+
+        FileWriter fileWriter = new FileWriter(FILE_NAME);
+
+        for (int i=0; i<Yappatron.taskArray.size(); i++){
+            fileWriter.write(Yappatron.taskArray.get(i).toString() + System.lineSeparator());
+        }
+        fileWriter.close();
+
+        }
+
+    }
+
+
```
try to be consistent in vertical spacing.

okkhoy commented on src/main/java/Taskmanager.java:
```diff
@@ -0,0 +1,81 @@
+    public void list(){
+        int i=0;
+        System.out.println("Here are the tasks in your list:");
+//        for (i=0; i<Yappatron.taskNumber; i++){
+//            System.out.print(i+1 + ". ");
+//            System.out.println(Yappatron.taskArray[i]);
+//        }
```
don't retain commented out code.

okkhoy commented on src/main/java/Taskmanager.java:
```diff
@@ -0,0 +1,81 @@
+            System.out.print(input + ". ");
+            System.out.println(task);
+            System.out.println("Now you have " + Yappatron.taskArray.size() + " tasks left");
+        }catch (IndexOutOfBoundsException e){
```
spacing problem

okkhoy reviewed (commented):
There are some coding standard violations. please try to fix them before the final submission.
You haven't packaged your code according to A-Packages

Commit:
Edit coding standard violations

Commit:
Create separate packages for classes
Separate functionalities into different classes (Parser, Ui, Taskmanager and FileManager)

Commit:
Merge branch 'edit-test'

Commit:
Edit attribute of Deadline class to handle LocalDateTime.
Edit addDeadline method to parse user input and create LocalDateTime object for deadline creation
Edit Parser to create Deadline object with LocalDateTime attribute when reading file

Commit:
Filter tasks using keyword
Print these filtered tasks
Edit Parser to return keyword to find
Edit Yappatron to include case for find command

ChuaCleon committed:
Merge pull request #1 from ChuaCleon/branch-Level-8

branch-level-8 pull

Commit:
Merge branch 'master' into branch-Level-9

ChuaCleon committed:
Merge pull request #2 from ChuaCleon/branch-Level-9

Branch level 9

Commit:
Add javadoc and header comments for classes and methods

ChuaCleon committed:
A-UserGuide

Commit:
Fix bug exiting program if invalid command typed

ChuaCleon committed:
Update README.md

Commit:
Add more javadoc comments

Commit:
Merge branch 'master' of https://github.com/ChuaCleon/ip

Commit:
Deal with DateTime exception when parsing dates and times for deadline

ChuaCleon committed:
Update README.md

Commit:
Update readme file
